ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/RemoteConnectorExecutor.java

```diff
@@ -41,7 +42,14 @@ default ModelTensorOutput executePredict(MLInput mlInput) {
                 if (tempTensorOutputs.size() > 0 && tempTensorOutputs.get(0).getMlModelTensors() != null) {
                     tensorCount = tempTensorOutputs.get(0).getMlModelTensors().size();
                 }
-                processedDocs += Math.max(tensorCount, 1);
+                // This is to support some model which takes N text docs and embedding size is less than N-1.
+                // We need to tell executor what's the step size for each model run.
+                Map<String, String> parameters = getConnector().getParameters();
+                if (parameters != null && parameters.containsKey("input_docs_processed_step_size")) {
+                    processedDocs += Integer.parseInt(parameters.get("input_docs_processed_step_size"));
```

Comment:
1. Is this standard param? Can different connector have different parameter name to represent step size?
2. What is the value returned is invalid int or out of range? (-ve value)

Reply:
1. This will be a reserved param for AI connector, just like the `region` for AWS region and `access_key` for AWS access key. All AI connector have to use this parameter to represent step size, just like they have to use `access_key` to represent AWS access key.
2. Good point, will add check logic, the step value should be greater than 0, not necessary to be less than input doc size.

Reply:
Addressed in this PR https://github.com/opensearch-project/ml-commons/pull/1587